docs/design/proposals/configure-snapshot-based-rbd-mirroring.md

```diff
@@ -0,0 +1,324 @@
+    ```
+
+* The secret `pool-peer-token-replicapool` contains all the information
+  related to the token and needs to be fed to the peer, but it is encoded.
+  To fetch the decoded secret:
+
+    ```sh
+    [cluster-b]$ kuberc get secrets pool-peer-token-replicapool --context=cluster-b -o jsonpath='{.data.token}'|base64 -d
+    ```
+
+* With this Decoded value, create a secret on the primary site(cluster-a),
+  using site name of the peer as the name.
+
+    ```sh
+    [cluster-a]$ kubectl -n rook-ceph create secret generic f4df4694-342f-4e77-9d1e-c97014a95818-rook-ceph --from-literal=token=eyJmc2lkIjoiZjRkZjQ2OTQtMzQyZi00ZTc3LTlkMWUtYzk3MDE0YTk1ODE4IiwiY2xpZW50X2lkIjoicmJkLW1pcnJvci1wZWVyIiwia2V5IjoiQVFDSThYNWZoRUtaTlJBQXdCYTdwekY2aTN4V0oyL2JPSmlBN3c9PSIsIm1vbl9ob3N0IjoiW3YyOjE5Mi4xNjguMzkuNDk6MzMwMCx2MToxOTIuMTY4LjM5LjQ5OjY3ODldIn0= --from-literal=pool=replicapool --context=cluster-a
+    ```
+
+    > Note: The command might be too long to directly run on cli,
+    > an alternate approach can be to save token in a file
+
+Now the kubernetes **secret is created on the primary site** (cluster-a),
+which contains token information required to connect with its peer i.e cluster-b.
+Now we have successfully registered cluster-b to cluster-a as a peer.
+Since it is two-way replication, **follow similar steps to on the peer to
+perform vice-versa**.
+
+## Create RBDMirror CRD
+
+Replication is handled by **rbd-mirror** daemon. The rbd-mirror daemon is responsible
+for pulling image updates from the remote, peer cluster, and applying them to image
+within the local cluster. Rook allows the creation and updating rbd-mirror daemon(s)
+through the custom resource definitions (CRDs).
+
+* Create `mirror.yaml`, to deploy the rbd-mirror daemon
+
+    ```yaml
+    apiVersion: ceph.rook.io/v1
+    kind: CephRBDMirror
+    metadata:
+      name: my-rbd-mirror
+      namespace: rook-ceph
+    spec:
+      # the number of rbd-mirror daemons to deploy
+      count: 1
+      peers:
+        secretNames:
+          # list of Kubernetes Secrets containing the peer token
+          - "f4df4694-342f-4e77-9d1e-c97014a95818-rook-ceph"
+    ```
+
+* Create the RBD mirror daemon
+
+  `[cluster-A]$ kubectl create -f mirror.yaml --context=cluster-a`
+
+* Validate if rbd-mirror daemon pod is now up
+
+  `[cluster-A]$ kuberc get pods --context=cluster-a`
+
+* Verify that daemon health is OK
+
+    ```sh
+    [cluster-A]$ kubectl get cephblockpool.ceph.rook.io/replicapool -nrook-ceph --context=cluster-a -oyaml
+    ```
+
+**Follow similar steps to on the peer to perform vice-versa.**
+
+## Enable Mirroring
+
+Once the rbd-mirror daemon is running on both the clusters, follow the below steps:
+
+* First, `ssh` into the toolbox pods of both clusters to check the pool status
+
+  `[cluster-a toolbox]$ rbd mirror pool status --pool=replicapool`
+  `[cluster-b toolbox]$ rbd mirror pool status --pool=replicapool`
+
+* Create an rbd image in cluster-a
```

Comment:
can you mention that this is done for testing, and that images are normally created by ceph-csi?